components/MaskedTransition/src/MDCMaskedTransition.m

```diff
@@ -0,0 +1,269 @@
+  MDCMaskedTransitionMotionSpec spec = motionForContext(context);
+  if (context.direction == MDMTransitionDirectionForward) {
+    _shouldSlideWhenCollapsed = spec.shouldSlideWhenCollapsed;
+  }
+
+  MDMMotionAnimator *animator = [[MDMMotionAnimator alloc] init];
+  animator.shouldReverseValues = context.direction == MDMTransitionDirectionBackward;
+
+  // Cache original state.
+  // We're going to reparent the fore view, so keep this information for later.
+  UIView *originalSuperview = context.foreViewController.view.superview;
+  const CGRect originalFrame = context.foreViewController.view.frame;
+  UIView *originalSourceSuperview = _sourceView.superview;
+  const CGRect originalSourceFrame = _sourceView.frame;
+  UIColor *originalSourceBackgroundColor = _sourceView.backgroundColor;
+
+  UIView *scrimView;
+  if (!_presentationController) {
+    scrimView = CreateScrimView(context);
+
+  } else {
+    scrimView = _presentationController.scrimView;
+  }
+
+  // The presentation controller, if available, will decide when to make the source view visible
+  // again.
+  _presentationController.sourceView = _sourceView;
+
+  // Reparent the fore view into a masked view.
+  UIView *maskedView = [[UIView alloc] initWithFrame:context.foreViewController.view.frame];
+  {
+    CGRect reparentedFrame = context.foreViewController.view.frame;
+    reparentedFrame.origin = CGPointZero;
+    context.foreViewController.view.frame = reparentedFrame;
+
+    maskedView.layer.cornerRadius = context.foreViewController.view.layer.cornerRadius;
+    maskedView.clipsToBounds = context.foreViewController.view.clipsToBounds;
+  }
+  [context.containerView addSubview:maskedView];
+
+  UIView *floodFillView = [[UIView alloc] initWithFrame:context.foreViewController.view.bounds];
+  floodFillView.backgroundColor = _sourceView.backgroundColor;
+
+  // TODO(featherless): Profile whether it's more performant to fade the flood fill out or to
+  // fade the fore view in (what we're currently doing).
+  [maskedView addSubview:floodFillView];
+  [maskedView addSubview:context.foreViewController.view];
+
+  // All frames are assumed to be relative to the container view unless named otherwise.
+  const CGRect initialSourceFrame = [_sourceView convertRect:_sourceView.bounds
+                                                      toView:context.containerView];
+  const CGRect finalMaskedFrame = originalFrame;
+  CGRect initialMaskedFrame;
+  CGPoint corner;
+  const CGPoint initialSourceCenter = CenterOfFrame(initialSourceFrame);
+  if (spec.isCentered) {
+    initialMaskedFrame = FrameCenteredAround(initialSourceCenter, originalFrame.size);
+    // Bottom right
+    corner = CGPointMake(CGRectGetMaxX(initialMaskedFrame), CGRectGetMaxY(initialMaskedFrame));
+
+  } else {
+    initialMaskedFrame = CGRectMake(context.containerView.bounds.origin.x,
+                                    initialSourceFrame.origin.y - 20,
+                                    originalFrame.size.width,
```

Comment:
originalFrame may not be standardized.
Use CGRectGetWidth()/Height()/MinY()/MinX.
If you decide to CGRectStandardize, please rename originalStandardizedFrame so this will not raise red flags.